nbconvert/templates/rst.tpl

```diff
@@ -9,7 +9,14 @@
 
 {% block input %}
 {%- if cell.source.strip() -%}
-.. code:: python
+{{".. code:: "-}}
+{%- if nb.metadata.language_info.pygments_lexer -%}
+    {{ nb.metadata.language_info.pygments_lexer }}
+{%- elif nb.metadata.language_info.codemirror_mode -%}
```

Comment:
I don't think this should try codemirror_mode; we use pygments_mode specifically because docutils, the main renderer of rst, uses pygments for highlighting. It's unlikely that Codemirror will be used, so the codemirror mode shouldn't have priority over the generic language name (also, codemirror mode can be a string or a dictionary, which confuses things more).